misc/ls-extensions/modules/project-design-service/src/main/java/io/ballerina/projectdesign/generators/GeneratorUtils.java

```diff
@@ -88,4 +99,48 @@ public static ServiceAnnotation getServiceAnnotation(NodeList<AnnotationNode> an
 
         return new ServiceAnnotation(id, label, elementLocation);
     }
+
+    public static ServiceAnnotation getServiceAnnotation(Annotatable annotableSymbol, String filePath) {
+
+        String id = null;
+        String label = "";
+        ElementLocation elementLocation = null;
+
+        List<AnnotationSymbol> annotSymbols = annotableSymbol.annotations();
+        List<AnnotationAttachmentSymbol> annotAttachmentSymbols = annotableSymbol.annotAttachments();
+        if (annotSymbols.size() == annotAttachmentSymbols.size()) {
+            for (int i = 0; i < annotSymbols.size(); i++) {
+                AnnotationSymbol annotSymbol = annotSymbols.get(i);
+                AnnotationAttachmentSymbol annotAttachmentSymbol = annotAttachmentSymbols.get(i);
+                String annotName = annotSymbol.getName().orElse("");
+                elementLocation = annotSymbol.getLocation().isPresent() ?
+                        getElementLocation(filePath, annotSymbol.getLocation().get().lineRange()) : null;
+                if (!(annotName.equals(DISPLAY_ANNOTATION) || annotAttachmentSymbol.attachmentValue().isEmpty()) ||
+                        !(annotAttachmentSymbol.attachmentValue().get().value() instanceof LinkedHashMap)) {
+                    continue;
+                }
+                LinkedHashMap attachmentValue = (LinkedHashMap) annotAttachmentSymbol.attachmentValue().get().value();
+                if (attachmentValue.containsKey(ID)) {
+                    id = (String) ((ConstantValue) attachmentValue.get(ID)).value();
+                }
+                if (attachmentValue.containsKey(LABEL)) {
+                    label = (String) ((ConstantValue) attachmentValue.get(LABEL)).value();
+                }
+                break;
+            }
+        }
+
+        return new ServiceAnnotation(id, label, elementLocation);
+    }
+
+    public static String getClientModuleName(Node clientNode, SemanticModel semanticModel) {
+
+        String clientModuleName = null;
+        Optional<TypeSymbol> clientTypeSymbol = semanticModel.typeOf(clientNode);
+        if (clientTypeSymbol.isPresent()) {
+            clientModuleName = clientTypeSymbol.get().signature().trim().replace(CLIENT, "");
+        }
+
+        return clientModuleName;
+    }
```

Comment:
better if we can return an `Optional<String>` in here, to avoid NPEs if the client name can't be extracted

Reply:
This method is a replica of https://github.com/ballerina-platform/ballerina-lang/blob/9066508d48c262064c2db956b0959a1b0c325897/misc/ls-extensions/modules/project-design-service/src/main/java/io/ballerina/projectdesign/generators/service/nodevisitors/StatementNodeVisitor.java#L113-L133 which also returns a null if the conditions fail, and I guess the only place this method is being used is to get the `connectorType`. and having null value for connector type is acceptable I guess.

Looping @aneeshafedo